crates/storage/provider/src/providers/static_file/manager.rs

```diff
@@ -114,6 +115,49 @@ impl StaticFileProvider {
         Self(Arc::new(provider))
     }
 
+    /// Reports metrics for the static files.
+    pub fn report_metrics(&self) -> ProviderResult<()> {
+        let Some(metrics) = &self.metrics else { return Ok(()) };
+
+        let static_files = iter_static_files(&self.path)?;
+        for (segment, ranges) in static_files {
+            let mut entries = 0;
+            let mut size = 0;
+
+            for (block_range, _) in &ranges {
+                let fixed_block_range = find_fixed_range(block_range.start());
+                let jar_provider = self
+                    .get_segment_provider(segment, || Some(fixed_block_range), None)?
+                    .ok_or(ProviderError::MissingStaticFileBlock(segment, block_range.start()))?;
+
+                entries += jar_provider.rows();
+
+                let data_size = File::open(jar_provider.data_path())
+                    .and_then(|file| file.metadata())
+                    .map(|metadata| metadata.len())
+                    .unwrap_or_default();
+                let index_size = File::open(jar_provider.index_path())
+                    .and_then(|file| file.metadata())
+                    .map(|metadata| metadata.len())
+                    .unwrap_or_default();
+                let offsets_size = File::open(jar_provider.offsets_path())
+                    .and_then(|file| file.metadata())
+                    .map(|metadata| metadata.len())
+                    .unwrap_or_default();
+                let config_size = File::open(jar_provider.config_path())
+                    .and_then(|file| file.metadata())
+                    .map(|metadata| metadata.len())
+                    .unwrap_or_default();
+
```

Comment:
could we just cache these for completed/frozen files?

Reply:
I think no static files can be considered frozen, because our pipeline can unwind any ranges, including the blocks in frozen jars.

There calls are relatively cheap, they're just a `stat` syscall to each of the paths.